packages/studio/src/components/StreamScopeInput.tsx

```diff
@@ -0,0 +1,78 @@
+import { useMemo } from "react";
+import useStudioStore from "../store/useStudioStore";
+import { StreamScope } from "@yext/studio-plugin";
+import StreamScopeField, {
+  StreamScopeFieldProps,
+} from "./common/StreamScopeField";
+import EntityIdField from "./AddPageButton/EntityIdField";
+
+export interface StreamScopeInputProps {
+  streamScope: StreamScope | undefined;
+  updateSelection: (
+    streamScopeField: keyof StreamScope
+  ) => (selectedIds: string[]) => void;
+}
+
+export default function StreamScopeInput(props: StreamScopeInputProps) {
+  const { streamScope, updateSelection } = props;
+  const streamScopeFields = useStreamScopeFields();
+  const totalStreamScopeItems = Object.values(streamScope ?? {}).reduce(
+    (numItems, scopeItems) => {
+      return numItems + scopeItems.length;
+    },
+    0
+  );
+
+  const multipleScopesSelected = !!(
+    (streamScope?.entityIds?.length && streamScope.entityTypes?.length) ||
+    (streamScope?.entityTypes?.length && streamScope.savedFilterIds?.length) ||
+    (streamScope?.savedFilterIds?.length && streamScope.entityIds?.length)
+  );
+
+  const filterIdOrEntityTypeSelected = !!(
+    streamScope?.entityTypes?.length || streamScope?.savedFilterIds?.length
+  );
+
+  return (
+    <>
+      <EntityIdField
```

Comment:
Currently, `useStreamScopeFileds()` only returns `entityTypes` and `savedFilterIds` since those are contained in the `accountContent` slice.  I think keeping the `<EntityIdField>` outside the map would probably be best, since I don't think we would want to add `entityIds` to the `streamScopeFields` array.

Reply:
I think that until we find a way to incorporate `EntityIdField` into `StreamScopeField` it makes the most sense to leave it as it is.  The structure of the data for `EntityIds` is different as compared to the `EntityTypes` or `SavedFilters` so it doesn't quite fit so nicely into the map without a little bit of weird data manipulation.

What are your thoughts?

Reply:
I'm fine leaving as is! I agree, that's what makes the most sense until we can find a nice way of incorporating `EntityIdField` into `StreamScopeField`.